sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientAsyncCodeSnippets.java

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+
+
+public class TableClientAsyncCodeSnippets {
+
+
+    public void AsyncSnippets() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        // Add a table
+        tableServiceAsyncClient.createTable("OfficeSupplies").subscribe(Void -> {
+            System.out.println("Table creation successful.");
+        }, error -> {
+            System.out.println("There was an error creating the table. Error: " + error);
+        });
+
+
+        // Delete a table
+        tableServiceAsyncClient.deleteTable("OfficeSupplies").subscribe(Void -> {
+            System.out.println("Table deletion successful");
+        }, error -> {
+            System.out.println("There was an error deleting the table. Error: " + error);
+        });
+
+
+        // Query tables
+        String filterString = "$filter=TableName eq 'OfficeSupplies'";
+        tableServiceAsyncClient.queryTables(null, null, filterString).subscribe(azureTable -> {
+            System.out.println(azureTable.getName());
+        }, error -> {
+            System.out.println("There was an error querying the service. Error: " + error);
+        });
+    }
+
+    public void InsertEntity() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getClient("OfficeSupplies");
+        String row = "crayolaMarkers";
+        String partitionKey = "markers";
+        HashMap<String, Object> tableEntityProperties = new HashMap<>();
```

Comment:
Is tableEntityProperties mandatory? If not, would we want an overload that doesn't take this HashMap?